### PR TITLE
fix: return appropriate error code when snapshot exists

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind bug
/kind failing-test

**What this PR does / why we need it**:
When creating the snapshot, we should return appropriate error code "AlreadyExists" if snapshot exists.

**Which issue(s) this PR fixes**:

Fixes #46 

**Special notes for your reviewer**:

**Release note**:
```
fix: return appropriate error code when snapshot exists
```

### CODE REVIEW

File: pkg/azuredisk/controllerserver.go [hunk 1]
```diff
@@ -461,7 +461,11 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	klog.V(2).Infof("begin to create snapshot(%s) under rg(%s)", snapshotName, d.cloud.ResourceGroup)
 	future, err := d.cloud.SnapshotsClient.CreateOrUpdate(ctx, d.cloud.ResourceGroup, snapshotName, snapshot)
 	if err != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("create snapshot error: %v", err))
+		if strings.Contains(err.Error(), "existing") {
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go]: what's the exact error? only matching `existing` word may be not enough

Comment (ZeroMagic) [reply pkg/azuredisk/controllerserver.go]: [CreateSnapshot] error(compute.SnapshotsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Changing property 'sourceResourceId' is not allowed for existing disk 'CreateSnapshot-snapshot-2'.")

Comment (andyzhangx) [reply pkg/azuredisk/controllerserver.go]: use `existing disk` word?

File: pkg/azuredisk/controllerserver.go [hunk 2]
```diff
@@ -464,7 +464,11 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	klog.V(2).Infof("begin to create snapshot(%s) under rg(%s)", snapshotName, d.cloud.ResourceGroup)
 	future, err := d.cloud.SnapshotsClient.CreateOrUpdate(ctx, d.cloud.ResourceGroup, snapshotName, snapshot)
 	if err != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("create snapshot error: %v", err))
+		if strings.Contains(err.Error(), "existing disk") {
+			return nil, status.Error(codes.AlreadyExists, fmt.Sprintf("the request snapshot already exists, but the SourceVolumeId is different"))
```
Comment (andyzhangx) [inline pkg/azuredisk/controllerserver.go]: `request snapshot(%s) under rg(%s) already exists, but the SourceVolumeId is different, error details: %v`
we should always show the error details and provide more details if possible.

Comment (ZeroMagic) [reply pkg/azuredisk/controllerserver.go]: done.

**Pull request merged**